Implemented: Made the new transfer order flow functional (#1302)

R-Sourabh opened the pull request:
### Related Issues
<!--  Put related issue number which this PR is closing. For example #123 -->

#1302


### Short Description and Why It's Useful
<!-- Describe in a few words what is this Pull Request changing and why it's useful -->

* Made the **Create Transfer Order** modal functional.
* Added the functionality to add items to the order from the **Create Transfer Order** page using scanning and searching.
* Updated the **TransferOrderItem** component to display items during order creation.
* In the same component, added functionality to update item quantities and remove items from the order after adding them.
* Added a footer button with options to discard the order, **Ship Later** (which approves the order and navigates to the order list page), and **Pack and Ship** (which navigates to the Ship Transfer Order page).
* On the **Ship Transfer Order** page, displayed the order items with their quantities on the left side, and a segment on the right side to either purchase a shipping label or manually add tracking. (Currently, only the manual tracking number functionality is working.)


### Screenshots of Visual Changes before/after (If There Are Any)
<!-- If you made any changes in the UI layer, please provide before/after screenshots -->


### Contribution and Currently Important Rules Acceptance
<!-- Please get familiar with following info -->

- [x] I read and followed [contribution rules](https://github.com/hotwax/fulfillment#contribution-guideline)

R-Sourabh committed:
Implemented: Improve Transfer Order Creation Flow UI (#1302)

R-Sourabh committed:
Merge branch 'main' into #1302-create-transfer-order-flow

R-Sourabh committed:
Removed: the unnecessary route from index.ts file

dt2patel committed:
review transfer order updated UI

dt2patel committed:
remove hr tag

dt2patel committed:
review ship transfer order UI

dt2patel committed:
update class name

R-Sourabh committed:
Improved: Updated the transfer order UI styling(#1302)

R-Sourabh committed:
Implemented: Made the new transfer order flow functional (#1302)

R-Sourabh committed:
Improved: resolved gemini commented reviews (#1302)

R-Sourabh committed:
Implemented: added support to update the order name and improved the change of statusFlowId (#1302)

R-Sourabh committed:
Improved: replaced order add/update item API with transfer order APIs (#1302)

R-Sourabh committed:
Improved: removed unused locale text and removed grandTotal dependency from the items in orders (#1302)

- Changed the fetchProductAverageCost service to fetch records for one product at a time, since we do not add products in bulk.

R-Sourabh committed:
Improved: Added the new order details api & separate api for order item details(#1302)

- Updated the payload for updating the order item and added the unitPrice field.
- Also added grandTotal field while creating the transfer order.

ravilodhi commented on src/components/TransferOrderItem.vue:
```diff
@@ -210,6 +222,51 @@ export default defineComponent({
+    async removeItemFromOrder(item: any) {
+      if(!item || !item.orderItemSeqId) return;
+      try {
+        const resp = await OrderService.deleteProductFromOrder({
```
Rename this to deleteOrderItem instead of deleteProductFromOrder.

ravilodhi commented on src/services/TransferOrderService.ts:
```diff
@@ -331,12 +347,62 @@ const approveTransferOrder = async (orderId: any): Promise<any> => {
   });
 };
 
+const addProductToOrder = async (payload: any): Promise<any> => {
```
This should be addItem or addOrderItem.

R-Sourabh committed:
Implemented: support for changing the order facility ID and added comments to various functions (#1302)

R-Sourabh committed:
Improved: changed the functions name & added shipping label functions in the ShipTransferOrder component (#1302)

ravilodhi commented on src/components/AddProductModal.vue:
```diff
@@ -0,0 +1,176 @@
+}
+
+// product search
+async function getProducts() {
```
Add scroll pagination for showing products, keep the pageSize to default set in env file. We already have similar behaviour in other Add/Search Product modal.

ravilodhi commented on src/components/AddProductModal.vue:
```diff
@@ -0,0 +1,176 @@
+}
+
+// Add product to order
+async function commitProductToOrder(product: any) {
```
Use naming like addItem or addOrderItem, no need to include "...ToOrder"

ravilodhi commented on src/components/AddProductModal.vue:
```diff
@@ -0,0 +1,176 @@
+  queryString.value = ''
+}
+
+async function addProductToOrderApi(newItem: any) {
```
Same here use, addItem or addOrderItem or addTransferOrderItem kind of naming.

R-Sourabh committed:
Implemented: added the data-testid to the actions for the create transfer order flow (#1302)

R-Sourabh committed:
Improved: updated the addTransferOrderItem function & updated the data-testids (#1302)

R-Sourabh committed:
Implemented: Added shipping workflows, enhance create transfer order component and added infinite scrolling in product modal(#130211)

- Improved AddProductModal with infinite scroll and better empty/loading states.
- Updated CreateTransferOrder to fix back button navigation, correct add-to-order logic, and ensure order status updates on approval.
- Refactored ShipTransferOrder to support label purchase, reprint, void, dynamic tracking URL generation, and improved handling of manual and purchased shipping flows.

ravilodhi commented on src/components/CreateTransferOrderModal.vue:
```diff
@@ -0,0 +1,187 @@
+const transferOrderName = ref('');
+const queryString = ref('');
+const facilities = ref([]) as any;
+const selectedDestinationFacilityid = ref('');
```
Correct the case of selectedDestinationFacilityid to selectedDestinationFacilityId
